### ISSUE TITLE
New Feature Offering: Compilers versions comparation

### ISSUE THREAD

**Issue by coralkashri** [andreasfertig/cppinsights#345]
I have a new cool feature to offer. In addition to choosing a compiler version, an ability to choose a versions comparison which will add a note to every line that will show which version was the last to modify this line by the compiler.
E.g.

Lambda expressions:

```cpp
#include <iostream>

int main() {  
  class __lambda_4_3 {
    public: 
    inline /*constexpr */ void operator()() const { // Since C++17
    }
    
    using retType_4_3 = void (*)();
    inline /*constexpr */ operator retType_4_3 () const noexcept {  // Since C++17
      return __invoke;
    };
    
    private: 
    static inline void __invoke() {
    }
    
    public:
    // /*constexpr */ __lambda_4_3() = default; // Since C++20
    
  } __lambda_4_3{};
  
  return 0;
}
```

**Comment by andreasfertig** [comment]
Hello @korelkashri,

thank you for thinking about this. I think there was another issue where, as a derail, we discussed having annotations. There are already minor things in C++ Insights, like saying that something is an NRVO variable. 

The way as you show it above, even I had to pause a few seconds first to get what is new. I think when annotations are added, there needs to be a mode to turn them on and off. It should also be clear to what an annotation belongs. In the above example, the first two annotations are regarding `constexpr` for the methods. In the last case, C++20, it is about the constructor being there. I think in the early beginnings, someone also suggested putting links in there to the Standard. All these are interesting things which bring a benefit but also add maintenance. Plus, it needs to be clear what is annotated.
I like to see more examples like yours, including alternative ways for how to annotate.

Andreas
